utils/obsproc/Ghrsst2Ioda.h

```diff
@@ -124,51 +122,57 @@ namespace gdasapp {
             mask[i][j] = 1;
           } else {
             mask[i][j] = 0;
           }
 
           // obs error
           // TODO(Somebody): add sampled std. dev. of sst to the total obs error
-          obserror[i][j] = (static_cast<float>(sstObsErr[0][i][j]) + errOffSet) * errScaleFactor;
+          obserror[i][j] = static_cast<float>(sstObsErr[index]) * errScaleFactor + errOffSet;
 
           // epoch time in seconds
-          seconds[i][j]  = static_cast<int64_t>((sstdTime[0][i][j] + dtimeOffSet)
-                                                * dtimeScaleFactor)
-                         + static_cast<int64_t>(refTime[0]);
+          seconds[i][j]  = static_cast<float>(sstdTime[index]) * dtimeScaleFactor
+                         + static_cast<float>(refTime[0]);
+          index++;
         }
       }
 
-      // TODO(Guillaume): check periodic BC, use sampling std dev of sst as a proxi for obs error
-      //                  should the sst mean be weighted by the provided obs error?
-      std::vector<std::vector<float>> lon2d_s =
-        gdasapp::superobutils::subsample2D(lon2d, mask, fullConfig_);
-      std::vector<std::vector<float>> lat2d_s =
-        gdasapp::superobutils::subsample2D(lat2d, mask, fullConfig_);
-      std::vector<std::vector<float>> sst_s =
-        gdasapp::superobutils::subsample2D(sst, mask, fullConfig_);
-      std::vector<std::vector<float>> obserror_s =
-        gdasapp::superobutils::subsample2D(obserror, mask, fullConfig_);
-      std::vector<std::vector<float>> seconds_s =
-        gdasapp::superobutils::subsample2D(seconds, mask, fullConfig_);
+      // Superobing
+      // TODO(Guillaume): Save the sampling std dev of sst so it can be used
+      //                  as a proxi for obs error
+      std::vector<std::vector<float>> sst_s;
+      std::vector<std::vector<float>> lon2d_s;
+      std::vector<std::vector<float>> lat2d_s;
+      std::vector<std::vector<float>> obserror_s;
+      std::vector<std::vector<float>> seconds_s;
+      if ( fullConfig_.has("binning") ) {
+        sst_s = gdasapp::superobutils::subsample2D(sst, mask, fullConfig_);
+        lon2d_s = gdasapp::superobutils::subsample2D(lon2d, mask, fullConfig_);
+        lat2d_s = gdasapp::superobutils::subsample2D(lat2d, mask, fullConfig_);
+        obserror_s = gdasapp::superobutils::subsample2D(obserror, mask, fullConfig_);
+        seconds_s = gdasapp::superobutils::subsample2D(seconds, mask, fullConfig_);
+      } else {
+        sst_s = sst;
+        lon2d_s = lon2d;
+        lat2d_s = lat2d;
+        obserror_s = obserror;
+        seconds_s = seconds;
+      }
 
       // number of obs after subsampling
       int nobs = sst_s.size() * sst_s[0].size();
 
       // Create instance of iodaVars object
       gdasapp::IodaVars iodaVars(nobs, {}, {});
 
-      // unix epoch at Jan 01 1981 00:00:00 GMT+0000
+      // Reference time is Jan 01 1981 00:00:00 GMT+0000
       iodaVars.referenceDate_ = refDate;
```

Comment:
The refDate from the input file is: "seconds since 1981-01-01 00:00:00".
The refDate for the  string dateTime:units should be "seconds since 1981-01-01T00:00:00Z", otherwise the UFO fails.
I have a fix that can be added...
      // Reference time is Jan 01 1981 00:00:00 GMT+0000
      // The string dateTime:units = "seconds since 1981-01-01T0:00:00Z"
      // Currentlythe refDate is "seconds since 1981-01-01 00:00:00"
      std:: string refDaten;
      refDaten.append(refDate.substr(0,24));
      refDaten.append("T");
      refDaten.append(refDate.substr(25,8));
      refDaten.append("Z");
      oops::Log::info() << refDaten << std::endl;
      iodaVars.referenceDate_ = refDaten;

Reply:
@guillaumevernieres @ShastriPaturi @AndrewEichmann-NOAA Hi all, I am still confused that ALL old python converters has ""seconds since 1970-01-01 00:00:00" as string dateTime:units,,, so do we need to match reference dTime as old python has had?

@ShastriPaturi I am not sure you mention reference Date "seconds since 1980-01-01 00:00:00". Am I thinking wrong?

Reply:
@apchoiCMD I found this issue when I ran the 3dvar_godas ctest.

Reply:
We're using the reference date from the file @apchoiCMD , which is 1980-01-01 . So yes, the datetime will be different.

Reply:
I fixed your bug @ShastriPaturi ...

Reply:
Thanks @guillaumevernieres